nixos/modules/services/misc/db-rest.nix

```
@@ -0,0 +1,187 @@
+{ config, pkgs, lib, options, ... }:
+let
+  inherit (lib) mkOption types mdDoc mkIf mkEnableOption literalExpression mkPackageOption maintainers;
+  cfg = config.services.db-rest;
+  opt = options.services.db-rest;
+in
+{
+  options = {
+    services.db-rest = {
+      enable = mkEnableOption (mdDoc "db-rest service");
+
+      user = mkOption {
+        type = types.str;
+        default = "db-rest";
+        description = mdDoc "User account under which db-rest runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "db-rest";
+        description = mdDoc "Group under which db-rest runs.";
+      };
+
+      host = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = mdDoc "The host address the db-rest server should listen on.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        default = 3000;
+        description = mdDoc "The port the db-rest server should listen on.";
+      };
+
+      redis = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = mdDoc "Enable caching with redis for db-rest.";
+        };
+
+        createLocally = mkOption {
+          type = types.bool;
+          default = true;
+          description = mdDoc "Configure a local redis server for db-rest.";
+        };
+
+        host = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = mdDoc "Redis host.";
+        };
+
+        port = mkOption {
+          type = with types; nullOr port;
+          default = null;
+          description = mdDoc "Redis port.";
+        };
+
+        user = mkOption {
+          type = with types; nullOr str;
+          default = null;
+          description = mdDoc "Optional username used for authentication with redis.";
+        };
+
+        passwordFile = mkOption {
+          type = with types; nullOr path;
+          default = null;
+          example = "/run/keys/db-rest/pasword-redis-db";
+          description = mdDoc "Path to a file containing the redis password.";
+        };
+
+        useSSL = mkOption {
+          type = types.bool;
+          default = true;
+          description = mdDoc "Use SSL if using a redis network connection.";
+        };
+      };
+
+      package = mkPackageOption pkgs "db-rest" { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.redis.createLocally && (cfg.redis.host != null || cfg.redis.port != null));
```

Maybe something like this would be more accurate?

```suggestion
        assertion = (cfg.redis.enable && !cfg.redis.createLocally) -> (cfg.redis.host != null && cfg.redis.port != null);
```